From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Thomas H.P. Andersen" <phomes@gmail.com>
Date: Tue, 20 Oct 2026 19:09:44 +0200
Subject: [PATCH] nvk: only upload descriptor state if dirty

push_dirty is not cleared after flush so it gets reuploaded
when it is not changed

For Deep Rock Galactic clearing it saves 30k reuploads per second

Reviewed-by: Mel Henning <mhenning@darkrefraction.com>
Reviewed-by: Mary Guillemard <mary@mary.zone>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/45452>
(cherry picked from commit 617f463e21a12a8125dae72f5b8818770da81195)
---
 src/nouveau/vulkan/nvk_cmd_buffer.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/nouveau/vulkan/nvk_cmd_buffer.c b/src/nouveau/vulkan/nvk_cmd_buffer.c
index d810686b..8b717fdd 100644
--- a/src/nouveau/vulkan/nvk_cmd_buffer.c
+++ b/src/nouveau/vulkan/nvk_cmd_buffer.c
@@ -1512,6 +1512,7 @@ nvk_cmd_buffer_flush_push_descriptors(struct nvk_cmd_buffer *cmd,
       };
       nvk_descriptor_state_set_root(cmd, desc, sets[set_idx], set_addr);
    }
+   desc->push_dirty = 0;
 }
 
 void
-- 
2.43.0

